### PR TITLE
Enable a couple unit tests for LinearAlgebra::TpetraWrappers::Vector

### PR DESCRIPTION
**Pull request by masterleinad**
We need to implement a `ReinitHelper` specialization, `locally_owned_domain_indices` and `locally_owned_range_indices` indices for these tests.

### CODE REVIEW

File: tests/lac/linear_operator_04.cc [hunk 1]
```diff
@@ -33,15 +34,28 @@ main(int argc, char *argv[])
...
+  {
+    TrilinosWrappers::SparseMatrix a;
+    auto op_a  = linear_operator<TrilinosWrappers::MPI::Vector>(a);
+    auto op_a2 = linear_operator<TrilinosWrappers::MPI::Vector>(a);
+  }
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+  {
+    TrilinosWrappers::SparseMatrix
+         a; // FIXME Should be LinearAlgebra::TepetraWrappers::SparseMatrix
```
Comment (bangerth) [inline tests/lac/linear_operator_04.cc:46]:
```suggestion
         a; // FIXME Should be LinearAlgebra::TpetraWrappers::SparseMatrix
```

**Review by bangerth — COMMENTED** [review]
Does that mean that we can multiply an Epetra matrix with Tpetra vectors?

**Comment by masterleinad** [comment]
> Does that mean that we can multiply an Epetra matrix with Tpetra vectors?

No, this test only reinitializes a Vector with a suitable parallel partitioning.

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
I'm ok to merge as is, but would appreciate if you could open an issue that tracks the missing function and that we want to update the test once the function has been added.

**Comment by masterleinad** [comment]
> I'm ok to merge as is, but would appreciate if you could open an issue that tracks the missing function and that we want to update the test once the function has been added.

I just added the missing the missing functions now.

**Review by bangerth — APPROVED** [review]
Even better!

**Pull request merged**